ProjectCoimbra.UWP/Project.Coimbra/Controls/InputControl.cs

```diff
@@ -2,491 +2,518 @@
+		/// animation, styling, binding, etc.
+		/// </summary>
+		public static readonly DependencyProperty PitchesProperty =
+			DependencyProperty.Register(
+				"Pitches",
+				typeof(IEnumerable<string>),
+				typeof(InputControl),
+				new PropertyMetadata(null, OnPitchesChanged));
+
+		/// <summary>
+		/// A <see cref="DependencyProperty"/> forming the backing store for the song title, to enable animation,
+		/// styling, binding, etc.
+		/// </summary>
+		public static readonly DependencyProperty SongTitleProperty =
+			DependencyProperty.Register("SongTitle", typeof(string), typeof(InputControl), new PropertyMetadata("Title"));
+
+		/// <summary>
+		/// A <see cref="DependencyProperty"/> forming the backing store for the time to next note counter, to enable animation,
+		/// styling, binding, etc.
+		/// </summary>
+		public static readonly DependencyProperty TimeToNextNoteProperty =
+			DependencyProperty.Register("TimeToNextNote", typeof(string), typeof(InputControl), new PropertyMetadata("TimeToNextNote"));
+
+		private readonly IList<Gamepad> gamepads = new List<Gamepad>(1);
+
+		private readonly object lockObject = new object();
+
+		private readonly Thread gamepadThread;
+
+		private ItemsControl buttons;
+
+		private ItemsControl pitchTracks;
+
+		private Grid pitchBackground;
+
+		private Canvas notes;
+
+		private Gamepad mainGamepad;
+
+		private Button[] buttonControls;
+
+		private TextBlock[] trackLabels;
+
+		private bool disposed;
+
+		/// <summary>
+		/// Initializes a new instance of the <see cref="InputControl"/> class.
+		/// </summary>
+		public InputControl()
+		{
+			this.DefaultStyleKey = typeof(InputControl);
+			this.SizeChanged += this.InputControl_SizeChanged;
+			this.Loaded += this.InputControl_Loaded;
+
+			Gamepad.GamepadAdded += this.Gamepad_GamepadAdded;
+			Gamepad.GamepadRemoved += this.Gamepad_GamepadRemoved;
+
+			this.gamepadThread = new Thread(this.PollGamepad);
+			this.gamepadThread.Start();
+		}
+
+		/// <summary>
+		/// The event invoked when the eye gaze control has been interacted with.
+		/// </summary>
+		public event EventHandler<StateChangedEventArgs> EyeGazeInteracted;
+
+		/// <summary>
+		/// The event invoked when a lane button has been clicked.
+		/// </summary>
+		public event EventHandler<int> LaneButtonClicked;
+
+		/// <summary>
+		/// Gets or sets the song title.
+		/// </summary>
+		public string SongTitle
+		{
+			get => (string)this.GetValue(SongTitleProperty);
+			set => this.SetValue(SongTitleProperty, value);
+		}
+
+		/// <summary>
+		/// Gets the song title.
+		/// </summary>
+		public string TimeToNextNote
+		{
+			get => (string)this.GetValue(TimeToNextNoteProperty);
+		}
+
+		/// <summary>
+		/// Sets the song title.
+		/// </summary>
+		public async void SetTimeToNextNote(string value)
+		{
+			await CoreApplication.MainView.CoreWindow.Dispatcher.RunAsync(CoreDispatcherPriority.Normal,
+				() =>
+				{
+					this.SetValue(TimeToNextNoteProperty, value);
+				});
```

Comment:
Please add `.ConfigureAwait(<Boolean>)` after this.

`false` is more efficient but may not work here, in which case use `true` to avoid issues that might occur if we upgrade to .NET Core.